subsys/net/lib/capture/capture.c

```diff
@@ -618,9 +618,9 @@ static int capture_send(const struct device *dev, struct net_if *iface,
 		return -ENOENT;
 	}
 
-	if (ctx->local.sa_family == AF_INET) {
+	if (IS_ENABLED(CONFIG_NET_IPV4) && ctx->local.sa_family == AF_INET) {
```

Comment:
No objection to adding `IS_ENABLED()`, I just did not see any linking issue if this is missing so the commit message is a bit confusing

Reply:
@jukkar 

To reproduce, I apply the following patch:
```
diff --git a/samples/net/sockets/http_client/prj.conf b/samples/net/sockets/http_client/prj.conf
index 985406f5b5d..48ea800dfdf 100644
--- a/samples/net/sockets/http_client/prj.conf
+++ b/samples/net/sockets/http_client/prj.conf
@@ -4,7 +4,7 @@ CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
 
 # Networking config
 CONFIG_NETWORKING=y
-CONFIG_NET_IPV4=y
+CONFIG_NET_IPV4=n
 CONFIG_NET_IPV6=y
 CONFIG_NET_TCP=y
 CONFIG_NET_SHELL=y
@@ -26,7 +26,7 @@ CONFIG_NET_CONTEXT_NET_PKT_POOL=y
 
 # Network address config
 CONFIG_NET_CONFIG_SETTINGS=y
-CONFIG_NET_CONFIG_NEED_IPV4=y
+CONFIG_NET_CONFIG_NEED_IPV4=n
 CONFIG_NET_CONFIG_NEED_IPV6=y
 CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.0.2.1"
 CONFIG_NET_CONFIG_MY_IPV4_GW="192.0.2.2"
@@ -46,3 +46,5 @@ CONFIG_NET_SOCKETS_LOG_LEVEL_DBG=n
 CONFIG_NET_HTTP_LOG_LEVEL_DBG=n
 CONFIG_NET_IPV6_LOG_LEVEL_DBG=n
 CONFIG_NET_IPV6_ND_LOG_LEVEL_DBG=n
+
+CONFIG_NET_CAPTURE=y
```

And build the http client sample.

Then I get:

```
.../subsys/net/lib/capture/capture.c:648: undefined reference to `net_if_ipv4_get_ttl'
```

Reply:
Indeed. I tried this earlier with the capture sample but probably missed some vital step as I did not see the same issue.